### ISSUE TITLE
Get rid of the warning

### ISSUE THREAD

**Issue by rgarrigue** [razorsedge/puppet-vmwaretools#54]
Hi

Is there a clean way to get rid of those warning ?

```
[root@ppbgows01 ~]# puppet apply --environment preproduction /etc/puppetlabs/code/environments/preproduction/manifests/
Warning: Unknown variable: '::vmwaretools_reposerver'. at /etc/puppetlabs/code/environments/preproduction/modules/vmwaretools/manifests/params.pp:21:17
Warning: Unknown variable: '::vmwaretools_repopath'. at /etc/puppetlabs/code/environments/preproduction/modules/vmwaretools/manifests/params.pp:26:15
Warning: Unknown variable: '::vmwaretools_repopriority'. at /etc/puppetlabs/code/environments/preproduction/modules/vmwaretools/manifests/params.pp:31:19
Warning: Unknown variable: '::vmwaretools_repoprotect'. at /etc/puppetlabs/code/environments/preproduction/modules/vmwaretools/manifests/params.pp:36:18
Warning: Unknown variable: '::vmwaretools_gpgkey_url'. at /etc/puppetlabs/code/environments/preproduction/modules/vmwaretools/manifests/params.pp:41:17
Warning: Unknown variable: '::vmwaretools_proxy'. at /etc/puppetlabs/code/environments/preproduction/modules/vmwaretools/manifests/params.pp:46:12
Warning: Unknown variable: '::vmwaretools_proxy_username'. at /etc/puppetlabs/code/environments/preproduction/modules/vmwaretools/manifests/params.pp:51:21
Warning: Unknown variable: '::vmwaretools_proxy_password'. at /etc/puppetlabs/code/environments/preproduction/modules/vmwaretools/manifests/params.pp:56:21
Warning: Unknown variable: '::vmwaretools_tools_version'. at /etc/puppetlabs/code/environments/preproduction/modules/vmwaretools/manifests/params.pp:61:20
Warning: Unknown variable: '::vmwaretools_ensure'. at /etc/puppetlabs/code/environments/preproduction/modules/vmwaretools/manifests/params.pp:68:13
Warning: Unknown variable: '::vmwaretools_package'. at /etc/puppetlabs/code/environments/preproduction/modules/vmwaretools/manifests/params.pp:73:14
```

Regards,

**Comment by alexjfisher** [comment]
@rgarrigue See https://github.com/razorsedge/puppet-vmwaretools/issues/55

**Comment by rgarrigue** [comment]
@alexjfisher Ok. Should I enforce `STRICT_VARIABLES=no` ? If I should, how do I do that, site.pp, puppet.conf ?

Thanks,

**Comment by alexjfisher** [comment]
It'd be `strict_variables = false` in puppet.conf on your master
https://docs.puppet.com/puppet/latest/reference/configuration.html#strictvariables

`STRICT_VARIABLES=no` is how you'd set the option when using rspec-puppet and puppetlabs_spec_helper. https://github.com/puppetlabs/puppetlabs_spec_helper#initializing-puppet-for-testing

I don't think this will help you though.  The option already defaults to `false` but I'm guessing in recent versions, puppet generates warnings.  If it was set to `true` I don't think the catalog would compile at all.

To fix the warnings, you'd have to fix the `params.pp` manifest in this module.  The fixes aren't hard, but there are quite a lot of them needed.  I *might* be able to find the time at some point, just probably not this week at least.

**Comment by rgarrigue** [comment]
So I tried the strict_variable=no, but as you guessed that's not changing
anything. Settings it to yes issue an error instead of a warning and stop
compilation, no is the default issue warning type.

So... well, that only cosmetic, nevermind this. Thanks for the answer
anyway !



2016-08-31 14:07 GMT+02:00 Alex Fisher <notifications@github.com>:

**Comment by razorsedge** [comment]
Fixed in #62 and released to the Forge as [6.1.0](https://forge.puppet.com/razorsedge/vmwaretools/6.1.0/).
